pkg/kubectl/sorting_printer.go

```diff
@@ -185,16 +186,37 @@ func isLess(i, j reflect.Value) (bool, error) {
 
 func (r *RuntimeSort) Less(i, j int) bool {
 	iObj := r.objs[i]
+	iElem := reflect.ValueOf(iObj).Elem()
 	jObj := r.objs[j]
+	jElem := reflect.ValueOf(jObj).Elem()
+	if r.field == "" && iElem.Type().Name() == "ReplicaSet" && jElem.Type().Name() == "ReplicaSet" {
```

Comment:
Added a comment - if we asked for a printer that was "list" aware, then the printer could observe that we got a list and invoke a dynamic call (that ran printReplicaSetList because it got a list of ReplicaSets).  Other types (Lists, groups of files from the CLI) would not get sorted or filtered, which is mostly what we want, because the user explicitly told us to get those items.